### ISSUE TITLE
[JENKINS-35820] Developer sees the log broken down by step

### ISSUE THREAD

**Issue by jenkins-infra-bot** [jenkinsci/blueocean-plugin#2788]
<p>In Scope</p>
<ul>
	<li>For any node in the railway the developer can click on it and see a list of the steps that had executed for that node <img class="emoticon" src="https://issues.jenkins.io/images/icons/emoticons/error.png" height="16" width="16" align="absmiddle" alt="" border="0"/></li>
	<li>Each step has a status (passed, failed or running)  <img class="emoticon" src="https://issues.jenkins.io/images/icons/emoticons/check.png" height="16" width="16" align="absmiddle" alt="" border="0"/></li>
	<li>Clicking on one of the steps expands the log in place. <img class="emoticon" src="https://issues.jenkins.io/images/icons/emoticons/check.png" height="16" width="16" align="absmiddle" alt="" border="0"/></li>
	<li>On successful, aborted, unstable and unknown runs the developer sees all the steps in the log collapsed <img class="emoticon" src="https://issues.jenkins.io/images/icons/emoticons/check.png" height="16" width="16" align="absmiddle" alt="" border="0"/></li>
	<li>On failed run the developer sees all the steps in the log collapsed but the failing step is expanded <img class="emoticon" src="https://issues.jenkins.io/images/icons/emoticons/check.png" height="16" width="16" align="absmiddle" alt="" border="0"/></li>
	<li>On executing run the developer sees all the steps in the log collapsed but the currently executing step is expanded <img class="emoticon" src="https://issues.jenkins.io/images/icons/emoticons/check.png" height="16" width="16" align="absmiddle" alt="" border="0"/></li>
	<li>Not in the mockup: 2px radius on the top and bottom un-expanded steps</li>
	<li>Don't fetch the whole log into the browser memory ever (break the "download log" feature if needed) <img class="emoticon" src="https://issues.jenkins.io/images/icons/emoticons/check.png" height="16" width="16" align="absmiddle" alt="" border="0"/></li>
	<li>Show full log in freestyle cases (no step by step break down) but without eagerly loading the whole multi MB log (should only load last 4K (ping Vivek), as jenkins console does now, with a link to "show full log" that then loads the rest. <img class="emoticon" src="https://issues.jenkins.io/images/icons/emoticons/warning.png" height="16" width="16" align="absmiddle" alt="" border="0"/> not test with other then pipeline and multibranch</li>
	<li>When loading a log (for a step, or whole log for freestyle) and it only loads the last part, offer a link to "Show all" (as Jenkins does now)</li>
</ul>

<p>Out of scope:</p>
<ul>
	<li>Tailing (<a href="https://issues.jenkins.io/browse/JENKINS-35841" title="Developer can tail the log of an executing stage" class="issue-link" data-issue-key="JENKINS-35841"><del>UX-73</del></a>)</li>
	<li>Download load option</li>
	<li>Open raw log in new tab (can lazily load from server) in all cases (reasonably formatter, nearly works already but missing newlines)</li>
</ul>

---
<details><summary><i>Originally reported by <img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/jamesdumay.png" title="jamesdumay's avatar" /> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=jamesdumay">jamesdumay</a>, imported from: <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-35820" target="_blank">Developer sees the log broken down by step</a></i></summary>
<i><ul>
<li><b>assignee</b>: <img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/tscherler.png" title="tscherler's avatar" /> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=tscherler">tscherler</a>
<li><b>status</b>: Closed
<li><b>priority</b>: Major
<li><b>component(s)</b>: blueocean-plugin
<li><b>label(s)</b>: blueocean-imported, japan
<li><b>resolution</b>: Fixed
<li><b>resolved</b>: 2016-06-15T08:31:00+00:00
<li><b>votes</b>: 1
<li><b>watchers</b>: 0
<li><b>imported</b>: 2025-12-01
</ul></i>
<details><summary>Raw content of original issue</summary>

<pre>
<p>In Scope</p>
<ul>
	<li>For any node in the railway the developer can click on it and see a list of the steps that had executed for that node <img class="emoticon" src="https://issues.jenkins.io/images/icons/emoticons/error.png" height="16" width="16" align="absmiddle" alt="" border="0"/></li>
	<li>Each step has a status (passed, failed or running)  <img class="emoticon" src="https://issues.jenkins.io/images/icons/emoticons/check.png" height="16" width="16" align="absmiddle" alt="" border="0"/></li>
	<li>Clicking on one of the steps expands the log in place. <img class="emoticon" src="https://issues.jenkins.io/images/icons/emoticons/check.png" height="16" width="16" align="absmiddle" alt="" border="0"/></li>
	<li>On successful, aborted, unstable and unknown runs the developer sees all the steps in the log collapsed <img class="emoticon" src="https://issues.jenkins.io/images/icons/emoticons/check.png" height="16" width="16" align="absmiddle" alt="" border="0"/></li>
	<li>On failed run the developer sees all the steps in the log collapsed but the failing step is expanded <img class="emoticon" src="https://issues.jenkins.io/images/icons/emoticons/check.png" height="16" width="16" align="absmiddle" alt="" border="0"/></li>
	<li>On executing run the developer sees all the steps in the log collapsed but the currently executing step is expanded <img class="emoticon" src="https://issues.jenkins.io/images/icons/emoticons/check.png" height="16" width="16" align="absmiddle" alt="" border="0"/></li>
	<li>Not in the mockup: 2px radius on the top and bottom un-expanded steps</li>
	<li>Don't fetch the whole log into the browser memory ever (break the "download log" feature if needed) <img class="emoticon" src="https://issues.jenkins.io/images/icons/emoticons/check.png" height="16" width="16" align="absmiddle" alt="" border="0"/></li>
	<li>Show full log in freestyle cases (no step by step break down) but without eagerly loading the whole multi MB log (should only load last 4K (ping Vivek), as jenkins console does now, with a link to "show full log" that then loads the rest. <img class="emoticon" src="https://issues.jenkins.io/images/icons/emoticons/warning.png" height="16" width="16" align="absmiddle" alt="" border="0"/> not test with other then pipeline and multibranch</li>
	<li>When loading a log (for a step, or whole log for freestyle) and it only loads the last part, offer a link to "Show all" (as Jenkins does now)</li>
</ul>

<p>Out of scope:</p>
<ul>
	<li>Tailing (<a href="https://issues.jenkins.io/browse/JENKINS-35841" title="Developer can tail the log of an executing stage" class="issue-link" data-issue-key="JENKINS-35841"><del>UX-73</del></a>)</li>
	<li>Download load option</li>
	<li>Open raw log in new tab (can lazily load from server) in all cases (reasonably formatter, nearly works already but missing newlines)</li>
</ul>

</pre>
</details>
</details>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/michaelneale.png" title="michaelneale's avatar" /> michaelneale:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-35820?focusedId=261221&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-261221">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>This was as <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=jdumay" class="user-hover" rel="jdumay">jdumay</a> and I discussed/designed last week. The log can also be downloaded/viewed for the whole job (likely in another tab). </p>

<p>By segmenting like this, we can also potentially speed up the page load if needed - we can load per flownode if needed. </p>
</pre>
</details>

</ul>
</details>

<p>This was as <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=jdumay" class="user-hover" rel="jdumay">jdumay</a> and I discussed/designed last week. The log can also be downloaded/viewed for the whole job (likely in another tab). </p>

<p>By segmenting like this, we can also potentially speed up the page load if needed - we can load per flownode if needed. </p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/michaelneale.png" title="michaelneale's avatar" /> michaelneale:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-35820?focusedId=261218&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-261218">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=vpandey" class="user-hover" rel="vpandey">vpandey</a> could you take a look at api - and let us know what could be done in the api to provide per step log info? 
Perhaps just having the log for a stage having a prefix for log lines per step is enough for us to collapse it together in the GUI - or else we need a step by step log - need advice please. </p>
</pre>
</details>

</ul>
</details>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=vpandey" class="user-hover" rel="vpandey">vpandey</a> could you take a look at api - and let us know what could be done in the api to provide per step log info? <br/>
Perhaps just having the log for a stage having a prefix for log lines per step is enough for us to collapse it together in the GUI - or else we need a step by step log - need advice please. </p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/michaelneale.png" title="michaelneale's avatar" /> michaelneale:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-35820?focusedId=261247&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-261247">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=jdumay" class="user-hover" rel="jdumay">jdumay</a> what do you think of a "scroll spy" type of thing (possibly not as initial pass) - as you scroll up or down, it expands the relevant section so you can smoothly navigate: <a href="http://www.w3schools.com/bootstrap/bootstrap_scrollspy.asp" class="external-link" target="_blank" rel="nofollow noopener">http://www.w3schools.com/bootstrap/bootstrap_scrollspy.asp</a></p>
</pre>
</details>

</ul>
</details>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=jdumay" class="user-hover" rel="jdumay">jdumay</a> what do you think of a "scroll spy" type of thing (possibly not as initial pass) - as you scroll up or down, it expands the relevant section so you can smoothly navigate: <a href="http://www.w3schools.com/bootstrap/bootstrap_scrollspy.asp" class="external-link" target="_blank" rel="nofollow noopener">http://www.w3schools.com/bootstrap/bootstrap_scrollspy.asp</a></p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>swashbuck1r:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-35820?focusedId=261393&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-261393">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>I love the collapse+expand sections.</p>

<p>Scrollspy is an interesting concept, but I suspect scrollspy with expanding sections could be a bit quirky &#8211; especially with short sections at the end.  It's more designed to track you in the TOC on long-form content.  For this, you'd have to choose a horizontal line in the window view (typically the top of the scroll view) that would expand the content, and then if you have short sections at the end, your scroll may never go down far enough to trigger the expansion of the bottom steps...so your smooth-scroll would stop and then you'd have to click on those bottom sections to open it.</p>

<p>For scroll spy you could also consider expanding all the content, and then use a TOC-style navigation on the side to click down to the sections.  That's more in the original design of scrollspy.  That would require pre-fetching all the content though &#8211; not sure if that's how its done today, or if you grab content at you expand.</p>



</pre>
</details>

</ul>
</details>

<p>I love the collapse+expand sections.</p>

<p>Scrollspy is an interesting concept, but I suspect scrollspy with expanding sections could be a bit quirky &#8211; especially with short sections at the end.  It's more designed to track you in the TOC on long-form content.  For this, you'd have to choose a horizontal line in the window view (typically the top of the scroll view) that would expand the content, and then if you have short sections at the end, your scroll may never go down far enough to trigger the expansion of the bottom steps...so your smooth-scroll would stop and then you'd have to click on those bottom sections to open it.</p>

<p>For scroll spy you could also consider expanding all the content, and then use a TOC-style navigation on the side to click down to the sections.  That's more in the original design of scrollspy.  That would require pre-fetching all the content though &#8211; not sure if that's how its done today, or if you grab content at you expand.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/jamesdumay.png" title="jamesdumay's avatar" /> jamesdumay:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-35820?focusedId=261479&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-261479">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=swashburn" class="user-hover" rel="swashburn">swashburn</a> great to hear you like that. We are tracking the scroll behaviour for this in the design ticket <a href="https://issues.jenkins.io/browse/JENKINS-35864" title="Step section in log sticks to result header when scrolling" class="issue-link" data-issue-key="JENKINS-35864"><del>UX-424</del></a>. Have you used Google Inbox? We are thinking about something very similar to their "sticky header" behaviour you see in long threads.</p>
</pre>
</details>

</ul>
</details>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=swashburn" class="user-hover" rel="swashburn">swashburn</a> great to hear you like that. We are tracking the scroll behaviour for this in the design ticket <a href="https://issues.jenkins.io/browse/JENKINS-35864" title="Step section in log sticks to result header when scrolling" class="issue-link" data-issue-key="JENKINS-35864"><del>UX-424</del></a>. Have you used Google Inbox? We are thinking about something very similar to their "sticky header" behaviour you see in long threads.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>swashbuck1r:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-35820?focusedId=261500&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-261500">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Sure, that inbox behavior makes sense at the very least.</p>
</pre>
</details>

</ul>
</details>

<p>Sure, that inbox behavior makes sense at the very least.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/michaelneale.png" title="michaelneale's avatar" /> michaelneale:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-35820?focusedId=261507&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-261507">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=swashburn" class="user-hover" rel="swashburn">swashburn</a> yeah - I think the scroll spy won't work here, I kind of liked it with the config page which is long but has tabs, but I don't think it works with the vertical collapsing steps - so can shelve that idea of "scroll spy" for now.</p>
</pre>
</details>

</ul>
</details>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=swashburn" class="user-hover" rel="swashburn">swashburn</a> yeah - I think the scroll spy won't work here, I kind of liked it with the config page which is long but has tabs, but I don't think it works with the vertical collapsing steps - so can shelve that idea of "scroll spy" for now.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/jamesdumay.png" title="jamesdumay's avatar" /> jamesdumay:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-35820?focusedId=261487&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-261487">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=swashburn" class="user-hover" rel="swashburn">swashburn</a> keep that feedback coming though <img class="emoticon" src="https://issues.jenkins.io/images/icons/emoticons/smile.png" height="16" width="16" align="absmiddle" alt="" border="0"/></p>
</pre>
</details>

</ul>
</details>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=swashburn" class="user-hover" rel="swashburn">swashburn</a> keep that feedback coming though <img class="emoticon" src="https://issues.jenkins.io/images/icons/emoticons/smile.png" height="16" width="16" align="absmiddle" alt="" border="0"/></p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/michaelneale.png" title="michaelneale's avatar" /> michaelneale:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-35820?focusedId=261462&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-261462">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Here is a slide deck showing the progression as stop-motion: </p>

<p><a href="https://docs.google.com/presentation/d/1jz_3mTH4rEldWlrHnm_04kHNtldItDhMXW2l3Ehqjio/edit#slide=id.g140c50d20d_0_0" class="external-link" target="_blank" rel="nofollow noopener">https://docs.google.com/presentation/d/1jz_3mTH4rEldWlrHnm_04kHNtldItDhMXW2l3Ehqjio/edit#slide=id.g140c50d20d_0_0</a>
(you will have to imagine meaningful logs loading via progressiveText). </p>

<p>However this is covered in <a href="https://issues.jenkins.io/browse/JENKINS-35841" title="Developer can tail the log of an executing stage" class="issue-link" data-issue-key="JENKINS-35841"><del>UX-73</del></a> and not in scope of this ticket.</p>
</pre>
</details>

</ul>
</details>

<p>Here is a slide deck showing the progression as stop-motion: </p>

<p><a href="https://docs.google.com/presentation/d/1jz_3mTH4rEldWlrHnm_04kHNtldItDhMXW2l3Ehqjio/edit#slide=id.g140c50d20d_0_0" class="external-link" target="_blank" rel="nofollow noopener">https://docs.google.com/presentation/d/1jz_3mTH4rEldWlrHnm_04kHNtldItDhMXW2l3Ehqjio/edit#slide=id.g140c50d20d_0_0</a><br/>
(you will have to imagine meaningful logs loading via progressiveText). </p>

<p>However this is covered in <a href="https://issues.jenkins.io/browse/JENKINS-35841" title="Developer can tail the log of an executing stage" class="issue-link" data-issue-key="JENKINS-35841"><del>UX-73</del></a> and not in scope of this ticket.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>svanoort:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-35820?focusedId=261230&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-261230">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>This is grand.  I love this accordian-style representation and the way it makes it easy to spot failures and see steps.</p>
</pre>
</details>

</ul>
</details>

<p>This is grand.  I love this accordian-style representation and the way it makes it easy to spot failures and see steps.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/jamesdumay.png" title="jamesdumay's avatar" /> jamesdumay:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-35820?focusedId=261227&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-261227">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=svanoort" class="user-hover" rel="svanoort">svanoort</a> boom! Glad you love it <img class="emoticon" src="https://issues.jenkins.io/images/icons/emoticons/smile.png" height="16" width="16" align="absmiddle" alt="" border="0"/></p>
</pre>
</details>

</ul>
</details>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=svanoort" class="user-hover" rel="svanoort">svanoort</a> boom! Glad you love it <img class="emoticon" src="https://issues.jenkins.io/images/icons/emoticons/smile.png" height="16" width="16" align="absmiddle" alt="" border="0"/></p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>svanoort:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-35820?focusedId=261251&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-261251">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=jdumay" class="user-hover" rel="jdumay">jdumay</a> It's a solid delighter for anyone who's had to look for the failure cause in parallel branches of a flow graph, and I do that pretty often.  Plus it looks really slick in the mockups.</p>
</pre>
</details>

</ul>
</details>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=jdumay" class="user-hover" rel="jdumay">jdumay</a> It's a solid delighter for anyone who's had to look for the failure cause in parallel branches of a flow graph, and I do that pretty often.  Plus it looks really slick in the mockups.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/tscherler.png" title="tscherler's avatar" /> tscherler:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-35820?focusedId=261338&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-261338">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>to be able to use ?start=xxx we need to get the X-TEXT response attribute to calculate backwards to e.g. start=X-TEXT-200. -&gt; UX-504</p>
</pre>
</details>

</ul>
</details>

<p>to be able to use ?start=xxx we need to get the X-TEXT response attribute to calculate backwards to e.g. start=X-TEXT-200. -> UX-504</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/tscherler.png" title="tscherler's avatar" /> tscherler:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-35820?focusedId=261511&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-261511">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=mneale" class="user-hover" rel="mneale">mneale</a> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=jdumay" class="user-hover" rel="jdumay">jdumay</a> it states: On failed run the developer sees all the steps in the log collapsed but the failing step is expanded </p>

<p>what happens if multiple steps fail? Should be the first visible or all?</p>

<p> <img src="https://issues.jenkins.io/images/icons/attach/noimage.png" imagetext="Screenshot from 2016-05-25 22-27-14.png|thumbnail" align="absmiddle" border="0" /> </p>
</pre>
</details>

</ul>
</details>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=mneale" class="user-hover" rel="mneale">mneale</a> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=jdumay" class="user-hover" rel="jdumay">jdumay</a> it states: On failed run the developer sees all the steps in the log collapsed but the failing step is expanded </p>

<p>what happens if multiple steps fail? Should be the first visible or all?</p>

<p> <img src="https://issues.jenkins.io/images/icons/attach/noimage.png" imagetext="Screenshot from 2016-05-25 22-27-14.png|thumbnail" align="absmiddle" border="0" /> </p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/tscherler.png" title="tscherler's avatar" /> tscherler:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-35820?focusedId=261524&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-261524">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>There is IMO a problem with the response the REST server sends for running jobs, I recorded the following on a pipeline that has finished the first stage and was running the second:</p>

<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-javascript"><span class="code-keyword">export</span> <span class="code-keyword">const</span> firstFinishedSecondRunning = [
  {
    <span class="code-quote">"_class"</span>: <span class="code-quote">"io.jenkins.blueocean.service.embedded.rest.PipelineNodeImpl"</span>,
    <span class="code-quote">"displayName"</span>: <span class="code-quote">"Stage 1"</span>,
    <span class="code-quote">"durationInMillis"</span>: 10264,
    <span class="code-quote">"edges"</span>: [
      {
        <span class="code-quote">"_class"</span>: <span class="code-quote">"io.jenkins.blueocean.service.embedded.rest.PipelineNodeImpl$EdgeImpl"</span>,
        <span class="code-quote">"id"</span>: <span class="code-quote">"12"</span>
      }
    ],
    <span class="code-quote">"id"</span>: <span class="code-quote">"5"</span>,
    <span class="code-quote">"result"</span>: <span class="code-quote">"SUCCESS"</span>,
    <span class="code-quote">"startTime"</span>: <span class="code-quote">"2016-05-25T13:47:40.534+0200"</span>,
    <span class="code-quote">"state"</span>: <span class="code-quote">"FINISHED"</span>
  },
  {
    <span class="code-quote">"_class"</span>: <span class="code-quote">"io.jenkins.blueocean.service.embedded.rest.PipelineNodeImpl"</span>,
    <span class="code-quote">"displayName"</span>: <span class="code-quote">"Stage 2"</span>,
    <span class="code-quote">"durationInMillis"</span>: 7271,
    <span class="code-quote">"edges"</span>: [
      {
        <span class="code-quote">"_class"</span>: <span class="code-quote">"io.jenkins.blueocean.service.embedded.rest.PipelineNodeImpl$EdgeImpl"</span>,
        <span class="code-quote">"id"</span>: <span class="code-quote">"16"</span>
      },
      {
        <span class="code-quote">"_class"</span>: <span class="code-quote">"io.jenkins.blueocean.service.embedded.rest.PipelineNodeImpl$EdgeImpl"</span>,
        <span class="code-quote">"id"</span>: <span class="code-quote">"17"</span>
      },
      {
        <span class="code-quote">"_class"</span>: <span class="code-quote">"io.jenkins.blueocean.service.embedded.rest.PipelineNodeImpl$EdgeImpl"</span>,
        <span class="code-quote">"id"</span>: <span class="code-quote">"18"</span>
      }
    ],
    <span class="code-quote">"id"</span>: <span class="code-quote">"12"</span>,
    <span class="code-quote">"result"</span>: <span class="code-quote">"UNKNOWN"</span>,
    <span class="code-quote">"startTime"</span>: <span class="code-quote">"2016-05-25T13:47:50.798+0200"</span>,
    <span class="code-quote">"state"</span>: <span class="code-quote">"RUNNING"</span>
  },
  {
    <span class="code-quote">"_class"</span>: <span class="code-quote">"io.jenkins.blueocean.service.embedded.rest.PipelineNodeImpl"</span>,
    <span class="code-quote">"displayName"</span>: <span class="code-quote">"firstBranch"</span>,
    <span class="code-quote">"durationInMillis"</span>: 7269,
    <span class="code-quote">"edges"</span>: [],
    <span class="code-quote">"id"</span>: <span class="code-quote">"16"</span>,
    <span class="code-quote">"result"</span>: <span class="code-quote">"SUCCESS"</span>,
    <span class="code-quote">"startTime"</span>: <span class="code-quote">"2016-05-25T13:47:50.800+0200"</span>,
    <span class="code-quote">"state"</span>: <span class="code-quote">"FINISHED"</span>
  },
  {
    <span class="code-quote">"_class"</span>: <span class="code-quote">"io.jenkins.blueocean.service.embedded.rest.PipelineNodeImpl"</span>,
    <span class="code-quote">"displayName"</span>: <span class="code-quote">"secondBranch"</span>,
    <span class="code-quote">"durationInMillis"</span>: 7268,
    <span class="code-quote">"edges"</span>: [],
    <span class="code-quote">"id"</span>: <span class="code-quote">"17"</span>,
    <span class="code-quote">"result"</span>: <span class="code-quote">"SUCCESS"</span>,
    <span class="code-quote">"startTime"</span>: <span class="code-quote">"2016-05-25T13:47:50.801+0200"</span>,
    <span class="code-quote">"state"</span>: <span class="code-quote">"FINISHED"</span>
  },
  {
    <span class="code-quote">"_class"</span>: <span class="code-quote">"io.jenkins.blueocean.service.embedded.rest.PipelineNodeImpl"</span>,
    <span class="code-quote">"displayName"</span>: <span class="code-quote">"thirdBranch"</span>,
    <span class="code-quote">"durationInMillis"</span>: 7267,
    <span class="code-quote">"edges"</span>: [],
    <span class="code-quote">"id"</span>: <span class="code-quote">"18"</span>,
    <span class="code-quote">"result"</span>: <span class="code-quote">"SUCCESS"</span>,
    <span class="code-quote">"startTime"</span>: <span class="code-quote">"2016-05-25T13:47:50.802+0200"</span>,
    <span class="code-quote">"state"</span>: <span class="code-quote">"FINISHED"</span>
  }
]
</pre>
</div></div>

<p>As you can see 16-18 are edge jobs and since 12 is currently running they should not have yet a result and the state should be RUNNING. May be related to <a href="https://cloudbees.atlassian.net/browse/UX-505?focusedCommentId=37907&amp;page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-37907" class="external-link" target="_blank" rel="nofollow noopener">https://cloudbees.atlassian.net/browse/UX-505?focusedCommentId=37907&amp;page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-37907</a>.</p>

<p>The real end result had been  <img src="https://issues.jenkins.io/images/icons/attach/noimage.png" imagetext="Screenshot from 2016-05-25 22-27-14.png|thumbnail" align="absmiddle" border="0" /> </p>
</pre>
</details>

</ul>
</details>

<p>There is IMO a problem with the response the REST server sends for running jobs, I recorded the following on a pipeline that has finished the first stage and was running the second:</p>


<pre>
<span class="code-keyword">export</span> <span class="code-keyword">const</span> firstFinishedSecondRunning = [
  {
    <span class="code-quote">"_class"</span>: <span class="code-quote">"io.jenkins.blueocean.service.embedded.rest.PipelineNodeImpl"</span>,
    <span class="code-quote">"displayName"</span>: <span class="code-quote">"Stage 1"</span>,
    <span class="code-quote">"durationInMillis"</span>: 10264,
    <span class="code-quote">"edges"</span>: [
      {
<span class="code-quote">"_class"</span>: <span class="code-quote">"io.jenkins.blueocean.service.embedded.rest.PipelineNodeImpl$EdgeImpl"</span>,
<span class="code-quote">"id"</span>: <span class="code-quote">"12"</span>
      }
    ],
    <span class="code-quote">"id"</span>: <span class="code-quote">"5"</span>,
    <span class="code-quote">"result"</span>: <span class="code-quote">"SUCCESS"</span>,
    <span class="code-quote">"startTime"</span>: <span class="code-quote">"2016-05-25T13:47:40.534+0200"</span>,
    <span class="code-quote">"state"</span>: <span class="code-quote">"FINISHED"</span>
  },
  {
    <span class="code-quote">"_class"</span>: <span class="code-quote">"io.jenkins.blueocean.service.embedded.rest.PipelineNodeImpl"</span>,
    <span class="code-quote">"displayName"</span>: <span class="code-quote">"Stage 2"</span>,
    <span class="code-quote">"durationInMillis"</span>: 7271,
    <span class="code-quote">"edges"</span>: [
      {
<span class="code-quote">"_class"</span>: <span class="code-quote">"io.jenkins.blueocean.service.embedded.rest.PipelineNodeImpl$EdgeImpl"</span>,
<span class="code-quote">"id"</span>: <span class="code-quote">"16"</span>
      },
      {
<span class="code-quote">"_class"</span>: <span class="code-quote">"io.jenkins.blueocean.service.embedded.rest.PipelineNodeImpl$EdgeImpl"</span>,
<span class="code-quote">"id"</span>: <span class="code-quote">"17"</span>
      },
      {
<span class="code-quote">"_class"</span>: <span class="code-quote">"io.jenkins.blueocean.service.embedded.rest.PipelineNodeImpl$EdgeImpl"</span>,
<span class="code-quote">"id"</span>: <span class="code-quote">"18"</span>
      }
    ],
    <span class="code-quote">"id"</span>: <span class="code-quote">"12"</span>,
    <span class="code-quote">"result"</span>: <span class="code-quote">"UNKNOWN"</span>,
    <span class="code-quote">"startTime"</span>: <span class="code-quote">"2016-05-25T13:47:50.798+0200"</span>,
    <span class="code-quote">"state"</span>: <span class="code-quote">"RUNNING"</span>
  },
  {
    <span class="code-quote">"_class"</span>: <span class="code-quote">"io.jenkins.blueocean.service.embedded.rest.PipelineNodeImpl"</span>,
    <span class="code-quote">"displayName"</span>: <span class="code-quote">"firstBranch"</span>,
    <span class="code-quote">"durationInMillis"</span>: 7269,
    <span class="code-quote">"edges"</span>: [],
    <span class="code-quote">"id"</span>: <span class="code-quote">"16"</span>,
    <span class="code-quote">"result"</span>: <span class="code-quote">"SUCCESS"</span>,
    <span class="code-quote">"startTime"</span>: <span class="code-quote">"2016-05-25T13:47:50.800+0200"</span>,
    <span class="code-quote">"state"</span>: <span class="code-quote">"FINISHED"</span>
  },
  {
    <span class="code-quote">"_class"</span>: <span class="code-quote">"io.jenkins.blueocean.service.embedded.rest.PipelineNodeImpl"</span>,
    <span class="code-quote">"displayName"</span>: <span class="code-quote">"secondBranch"</span>,
    <span class="code-quote">"durationInMillis"</span>: 7268,
    <span class="code-quote">"edges"</span>: [],
    <span class="code-quote">"id"</span>: <span class="code-quote">"17"</span>,
    <span class="code-quote">"result"</span>: <span class="code-quote">"SUCCESS"</span>,
    <span class="code-quote">"startTime"</span>: <span class="code-quote">"2016-05-25T13:47:50.801+0200"</span>,
    <span class="code-quote">"state"</span>: <span class="code-quote">"FINISHED"</span>
  },
  {
    <span class="code-quote">"_class"</span>: <span class="code-quote">"io.jenkins.blueocean.service.embedded.rest.PipelineNodeImpl"</span>,
    <span class="code-quote">"displayName"</span>: <span class="code-quote">"thirdBranch"</span>,
    <span class="code-quote">"durationInMillis"</span>: 7267,
    <span class="code-quote">"edges"</span>: [],
    <span class="code-quote">"id"</span>: <span class="code-quote">"18"</span>,
    <span class="code-quote">"result"</span>: <span class="code-quote">"SUCCESS"</span>,
    <span class="code-quote">"startTime"</span>: <span class="code-quote">"2016-05-25T13:47:50.802+0200"</span>,
    <span class="code-quote">"state"</span>: <span class="code-quote">"FINISHED"</span>
  }
]
</pre>

<p>As you can see 16-18 are edge jobs and since 12 is currently running they should not have yet a result and the state should be RUNNING. May be related to <a href="https://cloudbees.atlassian.net/browse/UX-505?focusedCommentId=37907&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-37907" class="external-link" target="_blank" rel="nofollow noopener">https://cloudbees.atlassian.net/browse/UX-505?focusedCommentId=37907&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-37907</a>.</p>

<p>The real end result had been  <img src="https://issues.jenkins.io/images/icons/attach/noimage.png" imagetext="Screenshot from 2016-05-25 22-27-14.png|thumbnail" align="absmiddle" border="0" /> </p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/tscherler.png" title="tscherler's avatar" /> tscherler:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-35820?focusedId=261422&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-261422">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=jdumay" class="user-hover" rel="jdumay">jdumay</a> an observation regarding the status symbols in front of the node log, we are introducing here a new set of status symbols, wouldn't it be better to reuse the ones from the activity view. This would IMO enhance the recognisability of the this statuses. I understand you are using the same as in the modalHeader, if we stay with that we should extract a new component from the modalHeader code generating the symbols and reuse them in the logNodes.</p>
</pre>
</details>

</ul>
</details>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=jdumay" class="user-hover" rel="jdumay">jdumay</a> an observation regarding the status symbols in front of the node log, we are introducing here a new set of status symbols, wouldn't it be better to reuse the ones from the activity view. This would IMO enhance the recognisability of the this statuses. I understand you are using the same as in the modalHeader, if we stay with that we should extract a new component from the modalHeader code generating the symbols and reuse them in the logNodes.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/tscherler.png" title="tscherler's avatar" /> tscherler:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-35820?focusedId=261495&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-261495">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p> <span class="error">&#91;^screencast_00002.mp4&#93;</span>  IMO if a job is failed and the last stage have not run it should not rotate, it will not be run never.</p>


<div class="code panel" style="border-width: 1px;"><div class="codeContent panelContent">
<pre class="code-javascript"><span class="code-keyword">export</span> <span class="code-keyword">const</span> runNodesFail = [{
    <span class="code-quote">'_class'</span>: <span class="code-quote">'io.jenkins.blueocean.service.embedded.rest.PipelineNodeImpl'</span>,
    displayName: <span class="code-quote">'Stage 1'</span>,
    durationInMillis: 10263,
    edges: [{
        <span class="code-quote">'_class'</span>: <span class="code-quote">'io.jenkins.blueocean.service.embedded.rest.PipelineNodeImpl$EdgeImpl'</span>,
        id: <span class="code-quote">'12'</span>
    }],
    id: <span class="code-quote">'5'</span>,
    result: <span class="code-quote">'SUCCESS'</span>,
    startTime: <span class="code-quote">'2016-05-24T13:42:07.833+0200'</span>,
    state: <span class="code-quote">'FINISHED'</span>
}, {
    <span class="code-quote">'_class'</span>: <span class="code-quote">'io.jenkins.blueocean.service.embedded.rest.PipelineNodeImpl'</span>,
    displayName: <span class="code-quote">'Stage 2'</span>,
    durationInMillis: 22141,
    edges: [{
        <span class="code-quote">'_class'</span>: <span class="code-quote">'io.jenkins.blueocean.service.embedded.rest.PipelineNodeImpl$EdgeImpl'</span>,
        id: <span class="code-quote">'15'</span>
    }, {
        <span class="code-quote">'_class'</span>: <span class="code-quote">'io.jenkins.blueocean.service.embedded.rest.PipelineNodeImpl$EdgeImpl'</span>,
        id: <span class="code-quote">'16'</span>
    }],
    id: <span class="code-quote">'12'</span>,
    result: <span class="code-quote">'FAILURE'</span>,
    startTime: <span class="code-quote">'2016-05-24T13:42:18.096+0200'</span>,
    state: <span class="code-quote">'FINISHED'</span>
}, {
    <span class="code-quote">'_class'</span>: <span class="code-quote">'io.jenkins.blueocean.service.embedded.rest.PipelineNodeImpl'</span>,
    displayName: <span class="code-quote">'firstBranch'</span>,
    durationInMillis: 22074,
    edges: [{
        <span class="code-quote">'_class'</span>: <span class="code-quote">'io.jenkins.blueocean.service.embedded.rest.PipelineNodeImpl$EdgeImpl'</span>,
        id: <span class="code-quote">'28'</span>
    }],
    id: <span class="code-quote">'15'</span>,
    result: <span class="code-quote">'SUCCESS'</span>,
    startTime: <span class="code-quote">'2016-05-24T13:42:18.098+0200'</span>,
    state: <span class="code-quote">'FINISHED'</span>
}, {
    <span class="code-quote">'_class'</span>: <span class="code-quote">'io.jenkins.blueocean.service.embedded.rest.PipelineNodeImpl'</span>,
    displayName: <span class="code-quote">'secondBranch'</span>,
    durationInMillis: 22076,
    edges: [{
        <span class="code-quote">'_class'</span>: <span class="code-quote">'io.jenkins.blueocean.service.embedded.rest.PipelineNodeImpl$EdgeImpl'</span>,
        id: <span class="code-quote">'28'</span>
    }],
    id: <span class="code-quote">'16'</span>,
    result: <span class="code-quote">'FAILURE'</span>,
    startTime: <span class="code-quote">'2016-05-24T13:42:18.099+0200'</span>,
    state: <span class="code-quote">'FINISHED'</span>
}, {
    <span class="code-quote">'_class'</span>: <span class="code-quote">'io.jenkins.blueocean.service.embedded.rest.PipelineNodeImpl'</span>,
    displayName: <span class="code-quote">'deploy'</span>,
    durationInMillis: <span class="code-keyword">null</span>,
    edges: [],
    id: <span class="code-quote">'28'</span>,
    result: <span class="code-keyword">null</span>,
    startTime: <span class="code-keyword">null</span>,
    state: <span class="code-keyword">null</span>
}];
</pre>
</div></div>
</pre>
</details>

</ul>
</details>

<p> <span class="error">&#91;^screencast_00002.mp4&#93;</span>  IMO if a job is failed and the last stage have not run it should not rotate, it will not be run never.</p>



<pre>
<span class="code-keyword">export</span> <span class="code-keyword">const</span> runNodesFail = [{
    <span class="code-quote">'_class'</span>: <span class="code-quote">'io.jenkins.blueocean.service.embedded.rest.PipelineNodeImpl'</span>,
    displayName: <span class="code-quote">'Stage 1'</span>,
    durationInMillis: 10263,
    edges: [{
<span class="code-quote">'_class'</span>: <span class="code-quote">'io.jenkins.blueocean.service.embedded.rest.PipelineNodeImpl$EdgeImpl'</span>,
id: <span class="code-quote">'12'</span>
    }],
    id: <span class="code-quote">'5'</span>,
    result: <span class="code-quote">'SUCCESS'</span>,
    startTime: <span class="code-quote">'2016-05-24T13:42:07.833+0200'</span>,
    state: <span class="code-quote">'FINISHED'</span>
}, {
    <span class="code-quote">'_class'</span>: <span class="code-quote">'io.jenkins.blueocean.service.embedded.rest.PipelineNodeImpl'</span>,
    displayName: <span class="code-quote">'Stage 2'</span>,
    durationInMillis: 22141,
    edges: [{
<span class="code-quote">'_class'</span>: <span class="code-quote">'io.jenkins.blueocean.service.embedded.rest.PipelineNodeImpl$EdgeImpl'</span>,
id: <span class="code-quote">'15'</span>
    }, {
<span class="code-quote">'_class'</span>: <span class="code-quote">'io.jenkins.blueocean.service.embedded.rest.PipelineNodeImpl$EdgeImpl'</span>,
id: <span class="code-quote">'16'</span>
    }],
    id: <span class="code-quote">'12'</span>,
    result: <span class="code-quote">'FAILURE'</span>,
    startTime: <span class="code-quote">'2016-05-24T13:42:18.096+0200'</span>,
    state: <span class="code-quote">'FINISHED'</span>
}, {
    <span class="code-quote">'_class'</span>: <span class="code-quote">'io.jenkins.blueocean.service.embedded.rest.PipelineNodeImpl'</span>,
    displayName: <span class="code-quote">'firstBranch'</span>,
    durationInMillis: 22074,
    edges: [{
<span class="code-quote">'_class'</span>: <span class="code-quote">'io.jenkins.blueocean.service.embedded.rest.PipelineNodeImpl$EdgeImpl'</span>,
id: <span class="code-quote">'28'</span>
    }],
    id: <span class="code-quote">'15'</span>,
    result: <span class="code-quote">'SUCCESS'</span>,
    startTime: <span class="code-quote">'2016-05-24T13:42:18.098+0200'</span>,
    state: <span class="code-quote">'FINISHED'</span>
}, {
    <span class="code-quote">'_class'</span>: <span class="code-quote">'io.jenkins.blueocean.service.embedded.rest.PipelineNodeImpl'</span>,
    displayName: <span class="code-quote">'secondBranch'</span>,
    durationInMillis: 22076,
    edges: [{
<span class="code-quote">'_class'</span>: <span class="code-quote">'io.jenkins.blueocean.service.embedded.rest.PipelineNodeImpl$EdgeImpl'</span>,
id: <span class="code-quote">'28'</span>
    }],
    id: <span class="code-quote">'16'</span>,
    result: <span class="code-quote">'FAILURE'</span>,
    startTime: <span class="code-quote">'2016-05-24T13:42:18.099+0200'</span>,
    state: <span class="code-quote">'FINISHED'</span>
}, {
    <span class="code-quote">'_class'</span>: <span class="code-quote">'io.jenkins.blueocean.service.embedded.rest.PipelineNodeImpl'</span>,
    displayName: <span class="code-quote">'deploy'</span>,
    durationInMillis: <span class="code-keyword">null</span>,
    edges: [],
    id: <span class="code-quote">'28'</span>,
    result: <span class="code-keyword">null</span>,
    startTime: <span class="code-keyword">null</span>,
    state: <span class="code-keyword">null</span>
}];
</pre>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/jamesdumay.png" title="jamesdumay's avatar" /> jamesdumay:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-35820?focusedId=261374&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-261374">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=tscherler" class="user-hover" rel="tscherler">tscherler</a> yes we are using the ones from the modal header right now. Circle filled with color vs status background color has to do with what the foreground contrast needs to be. If its white, on its own then we can use the circle variant. If we can't, its filled in and we use the modal header variant.</p>
</pre>
</details>

</ul>
</details>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=tscherler" class="user-hover" rel="tscherler">tscherler</a> yes we are using the ones from the modal header right now. Circle filled with color vs status background color has to do with what the foreground contrast needs to be. If its white, on its own then we can use the circle variant. If we can't, its filled in and we use the modal header variant.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/michaelneale.png" title="michaelneale's avatar" /> michaelneale:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-35820?focusedId=261249&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-261249">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=tscherler" class="user-hover" rel="tscherler">tscherler</a> use finished jobs for now, and then we may have to open other ticket for running states.</p>
</pre>
</details>

</ul>
</details>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=tscherler" class="user-hover" rel="tscherler">tscherler</a> use finished jobs for now, and then we may have to open other ticket for running states.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/jamesdumay.png" title="jamesdumay's avatar" /> jamesdumay:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-35820?focusedId=261448&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-261448">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=tscherler" class="user-hover" rel="tscherler">tscherler</a> </p>

<blockquote>
<p>On failed run the developer sees all the steps in the log collapsed but the failing step is expanded
what happens if multiple steps fail? Should be the first visible or all?</p></blockquote>
<p>If this is referring to the graph, the first failure in the graph should be visible.</p>
</pre>
</details>

</ul>
</details>

<p><a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=tscherler" class="user-hover" rel="tscherler">tscherler</a> </p>

<blockquote>
<p>On failed run the developer sees all the steps in the log collapsed but the failing step is expanded<br/>
what happens if multiple steps fail? Should be the first visible or all?</p></blockquote>
<p>If this is referring to the graph, the first failure in the graph should be visible.</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/michaelneale.png" title="michaelneale's avatar" /> michaelneale:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-35820?focusedId=261228&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-261228">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>FYI I just added an in-scope (<a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=jdumay" class="user-hover" rel="jdumay">jdumay</a>) as the log api is being corrected by <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=vpandey" class="user-hover" rel="vpandey">vpandey</a> to only load the last 4K<img class="emoticon" src="https://issues.jenkins.io/images/icons/emoticons/help_16.png" height="16" width="16" align="absmiddle" alt="" border="0"/> - there will be a need for a user to "show all" in cases where there is a lot of back-log they want to view (as is the case in jenkins now). </p>
</pre>
</details>

</ul>
</details>

<p>FYI I just added an in-scope (<a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=jdumay" class="user-hover" rel="jdumay">jdumay</a>) as the log api is being corrected by <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=vpandey" class="user-hover" rel="vpandey">vpandey</a> to only load the last 4K<img class="emoticon" src="https://issues.jenkins.io/images/icons/emoticons/help_16.png" height="16" width="16" align="absmiddle" alt="" border="0"/> - there will be a need for a user to "show all" in cases where there is a lot of back-log they want to view (as is the case in jenkins now). </p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/tscherler.png" title="tscherler's avatar" /> tscherler:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-35820?focusedId=261426&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-261426">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p><a href="https://github.com/jenkinsci/blueocean-plugin/pull/212/files#diff-ee06b67a788e43a3e9215a1940ab804fR546" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/blueocean-plugin/pull/212/files#diff-ee06b67a788e43a3e9215a1940ab804fR546</a> </p>

<p>We should simply change that to
`const logUrl = calculateLogUrl(config) + config.fullLog ? ' ?start=0' : '';`</p>
</pre>
</details>

</ul>
</details>

<p><a href="https://github.com/jenkinsci/blueocean-plugin/pull/212/files#diff-ee06b67a788e43a3e9215a1940ab804fR546" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/blueocean-plugin/pull/212/files#diff-ee06b67a788e43a3e9215a1940ab804fR546</a> </p>

<p>We should simply change that to<br/>
`const logUrl = calculateLogUrl(config) + config.fullLog ? ' ?start=0' : '';`</p>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i>sophistifunk:</i></summary>

<ul>
<li><i>Original <a class="no-jira-link-rewrite" href="https://issues.jenkins.io/browse/JENKINS-35820?focusedId=261445&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-261445">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>Layout-only component:</p>

<p> <img src="https://issues.jenkins.io/images/icons/attach/noimage.png" imagetext="Screen Shot 2016-06-10 at 3.29.44 PM.png|thumbnail" align="absmiddle" border="0" /> </p>
</pre>
</details>

</ul>
</details>

<p>Layout-only component:</p>

<p> <img src="https://issues.jenkins.io/images/icons/attach/noimage.png" imagetext="Screen Shot 2016-06-10 at 3.29.44 PM.png|thumbnail" align="absmiddle" border="0" /> </p>

**Comment by jenkins-infra-bot** [comment]
<i>[Original `blocks` from Jira: <a href="https://github.com/jenkinsci/blueocean-plugin/issues?q=is%3Aissue%20%22jira_issue_key%3DJENKINS-35841%22">JENKINS-35841</a>]</i>

**Comment by jenkins-infra-bot** [comment]
<i>[Original `blocks` from Jira: <a href="https://github.com/jenkinsci/blueocean-plugin/issues?q=is%3Aissue%20%22jira_issue_key%3DJENKINS-35822%22">JENKINS-35822</a>]</i>

**Comment by jenkins-infra-bot** [comment]
<i>[Original `blocks` from Jira: <a href="https://github.com/jenkinsci/blueocean-plugin/issues?q=is%3Aissue%20%22jira_issue_key%3DJENKINS-35886%22">JENKINS-35886</a>]</i>
